x-pack/platform/plugins/shared/onechat/server/routes/esql_tools.ts

```diff
@@ -0,0 +1,373 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { schema } from '@kbn/config-schema';
+import { esqlToolProviderId } from '@kbn/onechat-common';
+import { v4 as uuidv4 } from 'uuid';
+import type { RouteDependencies } from './types';
+import { getHandlerWrapper } from './wrap_handler';
+import { EsqlToolCreateRequest } from '../../common/tools';
+import { apiPrivileges } from '../../common/features';
+import { ESQL_TOOL_API_UI_SETTING_ID } from '../../common/constants';
+
+const TECHNICAL_PREVIEW_WARNING =
+  'Elastic ESQL Tool API is in technical preview and may be changed or removed in a future release. Elastic will work to fix any issues, but features in technical preview are not subject to the support SLA of official GA features.';
+
+export function registerESQLToolsRoutes({
+  router,
+  getInternalServices,
+  logger,
+}: RouteDependencies) {
+  const wrapHandler = getHandlerWrapper({ logger });
+
+  router.versioned
+    .get({
+      path: '/api/chat/tools/esql/{id}',
+      security: {
+        authz: { requiredPrivileges: [apiPrivileges.readOnechat] },
+      },
+      access: 'public',
+      summary: 'ESQL Tool API',
```

Comment:
Let's update the `summary` and `description` fields to reflect the actual functionality of each endpoint. Right now, they're all set to the same values. These fields might be used for API documentation generation, tbd.
@pquentin, do you know if we need to care about the `summary` and `description` values here?